lib/samples/show_legend/show_legend_sample.dart

```diff
@@ -0,0 +1,175 @@
+    );
+    // Create a feature layer.
+    final featureLayer = FeatureLayer.withFeatureTable(featureTable);
+    // Add image and feature layers to the operational layers list of the map.
+    _map
+      ..operationalLayers.add(imageLayer)
+      ..operationalLayers.add(featureLayer);
+    // Load the image and feature layers.
+    await featureLayer.load();
+    await imageLayer.load();
+    // Create a list to store operational layers and populate it with image and feature layers.
+    final operationalLayersList = <LayerContent>[
+      ...imageLayer.subLayerContents,
+      featureLayer
+    ];
+
+    // Get the legend info for each layer and add it to the legends dropdown list.
+    for (final layer in operationalLayersList) {
+      // Get the legend info for the current layer.
+      final layerLegends = await layer.fetchLegendInfos();
+      // Add the name of the current layer.
+      _legendsDropDown.add(
+        DropdownMenuItem(
+          value: layerLegends.first,
+          child: Text(
+            layer.name,
+            style: const TextStyle(
+              fontSize: 15,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+        ),
+      );
+
+      // Add current layer's legends to the dropdown list.
+      for (final legend in layerLegends) {
+        final ArcGISImage? arcGISImage;
+        // Create a swatch for the legend if the legend exists.
+        if (legend.symbol != null) {
+          arcGISImage = await legend.symbol!.createSwatch(
+            screenScale: screenScale,
+            backgroundColor: Colors.transparent,
+            size: const Size.square(6),
+          );
+        } else {
+          arcGISImage = null;
+        }
+
+        // Add the legend to the legends list.
+        _legendsDropDown.add(
+          DropdownMenuItem(
+            value: legend,
+            child: Row(
+              children: [
+                // Add the legend image to the dropdown list if the image exists.
+                arcGISImage != null
+                    ? Image.memory(
+                        arcGISImage.getEncodedBuffer(),
+                      )
+                    : Container(),
+                const SizedBox(width: 8),
+                // Add the legend name to the dropdown list.
+                Text(
+                  legend.name,
+                  style: const TextStyle(fontSize: 12),
+                ),
+              ],
+            ),
+          ),
+        );
+      }
+    }
+
+    // Set the map to the map view controller.
+    _mapViewController.arcGISMap = _map;
+    // Set the initial viewpoint of the map.
+    _map.initialViewpoint = Viewpoint.fromCenter(
+      ArcGISPoint(
+        x: -11e6,
+        y: 6e6,
+        spatialReference: SpatialReference.webMercator,
+      ),
+      scale: 9e7,
+    );
+    // Reset the state once the dropdown list is updated.
+    setState(() {});
```

Comment:
I would put this right after the big block of code that builds `_legendsDropDown`, to make it a little more clear what it's doing.

Actually, another idea would be to declare a local temporary variable, `final legendsDropDown = <DropdownMenuItem<LegendInfo>>[]`, and use that one to build up the list. Then, after the whole list is built, do a call to `setState` like:
```dart
setState(() => _legendsDropDown = legendsDropDown);
```
That way, we don't have to worry about the list getting built a little at a time with "async" jumps in between. I'm worried that the `build()` method might get called while the dropdown list is only partially built.

Reply:
Yes that sounds like a good idea - we would also avoid having an empty setState()